aurora/src/luks-unlock.py

```
@@ -0,0 +1,450 @@
+
+    while time.time() < deadline:
+        # Primary path: detect Plymouth passphrase prompt via serial log.
+        # With console=tty0 console=ttyS0 in the BLS entry, Plymouth writes
+        # the prompt to serial.  Input still comes from tty0, so sendkey works.
+        serial_result = qemu_check_serial(serial_log)
+        if serial_result == "plymouth":
+            print("[luks-unlock] Plymouth passphrase prompt detected via serial", flush=True)
+            brightness, md5 = qemu_screendump(monitor_sock, snap)
+            try:
+                import shutil
+                shutil.copy2(snap, "/tmp/luks-screenshot-plymouth.ppm")
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/luks-screenshot-plymouth.ppm"

(S108)